modules/orientation/src/main/java/com/gluonhq/attach/orientation/impl/AndroidOrientationService.java

```diff
@@ -0,0 +1,82 @@
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL GLUON BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.gluonhq.attach.orientation.impl;
+
+import com.gluonhq.attach.orientation.OrientationService;
+import java.util.Optional;
+import javafx.application.Platform;
+import javafx.beans.property.ReadOnlyObjectProperty;
+import javafx.beans.property.ReadOnlyObjectWrapper;
+import javafx.geometry.Orientation;
+
+public class AndroidOrientationService implements OrientationService {
+
+    static {
+        System.loadLibrary("Orientation");
+    }
+
+    private static ReadOnlyObjectWrapper<Orientation> orientation;
+    private static String orientationText = "Unknown";
+
+    public AndroidOrientationService() {
+        orientation = new ReadOnlyObjectWrapper<>();
+    }
+
+    @Override
+    public ReadOnlyObjectProperty<Orientation> orientationProperty() {
+        return orientation.getReadOnlyProperty();
+    }
+    
+    @Override
+    public final Optional<Orientation> getOrientation() {
+        return convertOrientation(orientationText);
+    }
+    
+    // callback
+    private static void notifyOrientation(String o) {
+        convertOrientation(o).ifPresent(or -> {
+            if (orientation.get() == null || !orientation.get().equals(or)) {
+                Platform.runLater(() -> orientation.setValue(or));
+            }
+        });
+    }
+
+    private static Optional<Orientation> convertOrientation(String text) {
+        orientationText = text;
+        switch (orientationText) {
+            case "Portrait":
+                return Optional.of(Orientation.VERTICAL);
+            case "Landscape":
+                return Optional.of(Orientation.HORIZONTAL);
+            case "Unknown":
```

Comment:
"Unknown" is already handled by the default case.

Reply:
Indeed, but I've added it there, so we explicitly match the strings we send from the dalvik side.